### PR TITLE
Create KMPUseCases Mocks

### PR DESCRIPTION
**Pull request by DavidKadlcek**
# :pencil: Description
- There was a problem to mock KMP UseCases so I resolved it 😈 Firstly we have to create a class of the mock inside the `DevstackKmpSharedMocks` where you define your conformations. Then you can use the mock inside your tests. I create the example of the test for `BooksViewModel` -> `BooksViewModelTests`

# :bulb: What’s new?
- Add helper classes - `UseCaseResultMock`, `UseCaseResultNoParamsMock`,  `UseCaseFlowNoParamsMock<Out>`, `TestError`

# :no_mouth: What’s missing?
- Nothing

# :books: References
- Technical calls

### CODE REVIEW

File: ios/DomainLayer/SharedDomain/Sources/SharedDomainMocks/Extensions/UseCaseStreamStreamMock.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
+//
+//  Created by David Kadlček on 27.02.2023
+//  Copyright © 2023 Matee. All rights reserved.
+//
+
+import Foundation
+import DevstackKmpShared
+
+open class UseCaseFlowNoParamsMock<Out>: UseCaseFlowNoParams {
```
Comment (pchmelar) [inline ios/DomainLayer/SharedDomain/Sources/SharedDomainMocks/Extensions/UseCaseStreamStreamMock.swift]: Wrong file name `UseCaseStreamStreamMock` .. should be `UseCaseFlowNoParamsMock`
Also, I'm not sure about the folder naming, these are not extensions, so maybe name it something like `KMPUtilities`? And you can put that `TestError` there as well 🙂

File: ios/PresentationLayer/Recipes/Tests/RecipesTests/BooksViewModelTests.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import Resolver
+import SharedDomain
+import SharedDomainMocks
+import DevstackKmpShared
+import UIToolkit
+import XCTest
+
+@MainActor
+final class BooksViewModelTests: XCTestCase {
+
```
Comment (pchmelar) [inline ios/PresentationLayer/Recipes/Tests/RecipesTests/BooksViewModelTests.swift:18]: pls `// MARK: Dependencies` to keep it consistent with other tests

File: ios/PresentationLayer/Recipes/Tests/RecipesTests/BooksViewModelTests.swift [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        registerKMPMocks()
+        
+        return BooksViewModel(flowController: fc)
+    }
+    
+    private func registerKMPMocks() {
+        Resolver.register { self.getBooksUseCaseMock as GetBooksUseCase }
+        Resolver.register { self.refreshBooksUseCaseMock as RefreshBooksUseCase }
+    }
+
```
Comment (pchmelar) [inline ios/PresentationLayer/Recipes/Tests/RecipesTests/BooksViewModelTests.swift:36]: pls `// MARK: Tests` to keep it consistent with other tests

**Review by pchmelar — CHANGES_REQUESTED** [review]

File: ios/PresentationLayer/Recipes/Tests/RecipesTests/BooksViewModelTests.swift (hunk 3 above)
Comment (DavidKadlcek) [reply ios/PresentationLayer/Recipes/Tests/RecipesTests/BooksViewModelTests.swift:36]: Done

File: ios/PresentationLayer/Recipes/Tests/RecipesTests/BooksViewModelTests.swift (hunk 2 above)
Comment (DavidKadlcek) [reply ios/PresentationLayer/Recipes/Tests/RecipesTests/BooksViewModelTests.swift:18]: Done

File: ios/DomainLayer/SharedDomain/Sources/SharedDomainMocks/Extensions/UseCaseStreamStreamMock.swift (hunk 1 above)
Comment (DavidKadlcek) [reply ios/DomainLayer/SharedDomain/Sources/SharedDomainMocks/Extensions/UseCaseStreamStreamMock.swift]: Done

**Review by pchmelar — APPROVED** [review]

**Pull request merged**